scalanet/discovery/src/io/iohk/scalanet/discovery/ethereum/v4/DiscoveryNetwork.scala

```diff
@@ -385,6 +385,10 @@ object DiscoveryNetwork {
                 // New response, fold it with the existing to decide if we need more.
                 val next = (acc: Z) => Some(acc -> (count + 1))
                 Task.pure(f(acc, response).bimap(next, next))
+
+              case _ => 
+                // Invalid state - this cannot happen
+                Task(logger.debug(s"Unexpected state while collecting responses from ${channel.to}")).as(Right(None))
```

Comment:
```suggestion
              case (None, _) => 
                Task.raiseError(new IllegalStateException(s"Unexpected state while collecting responses from ${channel.to}"))
```
So it doesn't go unnoticed.